packages/twenty-front/src/modules/object-record/record-board/hooks/internal/useRecordBoardCardFieldsInternal.ts

```diff
@@ -39,25 +39,20 @@ export const useRecordBoardCardFieldsInternal = (
           .getLoadable(recordBoardCardFieldsScopedState({ scopeId }))
           .getValue();
 
-        const existingFieldsUpdated = existingFields.map((previousField) =>
-          previousField.fieldMetadataId === field.fieldMetadataId
-            ? { ...previousField, isVisible: !field.isVisible }
-            : previousField,
-        );
-
-        const isNewField = !existingFields.find(
+        const fieldIndex = existingFields.findIndex(
           ({ fieldMetadataId }) => field.fieldMetadataId === fieldMetadataId,
         );
-
-        const fields = isNewField
-          ? [
-              ...existingFieldsUpdated,
-              {
-                ...field,
-                position: existingFieldsUpdated.length,
-              },
-            ]
-          : existingFieldsUpdated;
+        const fields = [...existingFields];
+
+        if (fieldIndex === -1) {
+          fields.push({ ...field, position: existingFields.length });
+        } else {
+          fields[fieldIndex] = {
+            ...field,
+            isVisible: !field.isVisible,
+            position: existingFields.length,
```

Comment:
Not sure we should be changing the field's position 🤔

Reply:
this is the existing behavior!